src/blog/system-analyzer.md

```diff
@@ -0,0 +1,204 @@
+
+We want to know why we are creating multiple Map shapes for the same type of objects. To do so, we toggle the info button about IC state to get more information about the Map addresses going from uninitialised to monomorphic.
+
+![The map transition tree associated with the monomorphic IC state.](/_img/system-analyzer/case1_2.png)
+
+![The map transition tree associated with the polymorphic IC state.](/_img/system-analyzer/case1_3.png)
+
+For the monomorphic IC state we can visualise the transition tree and see that we are only dynamically adding two properties `x`and `y` but when it comes to polymorphic IC state, we have a new Map containing three properties `isNegative`, `x` and `y`.
+
+![The Map panel communicates the file position information to highlight file positions on the Source panel.](/_img/system-analyzer/case1_4.png)
+
+We click on the file position section of the Map panel to see where this `isNegative` property is added in the source code and can use this insight to address the performance regression.
+
+So now the question being *how can we address the performance regression by using the insight we generated from the tool*?
+
+The minimal solution would be to always initialise the `isNegative` property. In general, it is sound advice that all instance properties should be initialised in the constructor.
+
+Now, the updated `Point` class looks like this:
+
+```javascript
+class Point {
+ constructor(x, y) {
+   this.isNegative = x < 0 || y < 0;
+   this.x = x;
+   this.y = y;
+ }
+
+ dotProduct(other) {
+   return this.x * other.x + this.y * other.y;
+ }
+}
+```
+
+If we execute the script again with the modified `Point` class, we see that the execution of the two snippets defined at the beginning of the case study perform very similarly.
+
+In an updated trace, we see that the polymorphic IC state is avoided as we are not creating multiple maps for the same type of objects.
+
+![The map transition tree of the modified Point object.](/_img/system-analyzer/case2_1.png)
+
+## The System Analyzer
+
+Let's now have an in-depth look at the different panels that are present in the system analyzer.
+
+### Timeline Panel
+
+The Timeline panel allows selection in time which enables visualization of IC/map states across discrete points in time or a selected range in time. It supports filtering features such as zoom in/out to the log events for selected time ranges.
+
+![Timeline panel overview](/_img/system-analyzer/timeline-panel.png)
+
+![Timeline panel overview (Cont.)](/_img/system-analyzer/timeline-panel2.png)
+
+### Map Panel
+
+The Map panel has two sub panels:
+
+1. Map details
+
+2. Map transitions
```

Comment:
Please remove a newline above, otherwise it tries to render them as paragraphs within a list or something like that.

Reply:
Looks like I need these empty lines: https://github.com/v8/v8.dev/pull/486/checks?check_run_id=1193656862 ?

Reply:
You need an empty line before the list, but not between items.